spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/env/EnvironmentEndpointAutoConfiguration.java

```diff
@@ -48,13 +55,19 @@ public EnvironmentEndpointAutoConfiguration(
 	@Bean
 	@ConditionalOnMissingBean
 	@ConditionalOnEnabledEndpoint
-	public EnvironmentEndpoint environmentEndpoint(Environment environment) {
-		EnvironmentEndpoint endpoint = new EnvironmentEndpoint(environment);
+	public EnvironmentEndpoint environmentEndpoint(Environment environment,
+			ObjectProvider<Collection<EnvironmentEndpoint.SanitizeFilter>> customFilters) {
+
 		String[] keysToSanitize = this.properties.getKeysToSanitize();
-		if (keysToSanitize != null) {
-			endpoint.setKeysToSanitize(keysToSanitize);
-		}
-		return endpoint;
+		KeyBlacklistSanitizeFilter keyMatchFilter = (keysToSanitize != null)
+				? KeyBlacklistSanitizeFilter.matchingKeys(keysToSanitize)
+				: KeyBlacklistSanitizeFilter.matchingDefaultKeys();
```

Comment:
I wanted to create the instance of `KeyBlacklistSanitizeFilter` as a bean that we would autowire here, but couldn't express the condition "register the filter bean only if environment endpoint is enabled". Is there a way for expressing that?